### PR TITLE
Update runtime to FreeDesktop 23.08

### PR DESCRIPTION
**Pull request by prodrigestivill**
I created this PR to make this changes:

- Update freedesktop runtime to 23.08
- ~Update to OpenJDK 17 (since version 11 EOL at the end of this month)~
- ~Add automatic PR creations by using [flathub external checker](https://docs.flathub.org/docs/for-app-authors/external-data-checker/).~
- ~Run flathub external checker to validate the configuration and commit the changes it made.~

Please let me know if you prefer separated PR, or to add or remove any of the changes for this PR.

Regarding the flathub external checker configuration, the flathub bot will create PR on each new zaproxy release, an option can be configured to automatically merge those PR by changing the file `flathub.json` by something like this:

```json
{
  "only-arches": ["x86_64"],
  "automerge-flathubbot-prs": true
}
```

Let me know also if you want me to add this change in this PR.

### CODE REVIEW

**Comment by kingthorin** [comment]
Not that I disagree, I have no problem moving to 17. But active support ends in a little over two weeks, security support continues for three years: https://endoflife.date/java

**Comment by psiinon** [comment]
Thanks for this PR!
However we should not move to Java 17 until we have a new ZAP release.
Java 17 does not include Nashorn, and the current version of ZAP has some problems with GraalJS which is the replacement we are suing.
The next version of ZAP should fix these problems so we can update to Java 17 then.

**Comment by prodrigestivill** [comment]
ok, no problem.
Just removed OpenJDK update.

File: org.zaproxy.ZAP.json [hunk 1]
```diff
@@ -43,7 +45,14 @@
                 {
                     "type": "archive",
                     "url": "https://github.com/zaproxy/zaproxy/releases/download/v2.13.0/ZAP_2.13.0_Linux.tar.gz",
-                    "sha256": "936eb52a0fd390c1ef890c455420d95ce20062fe136ec0927e023e2baf50f549"
+                    "sha256": "936eb52a0fd390c1ef890c455420d95ce20062fe136ec0927e023e2baf50f549",
+                    "x-checker-data": {
```
Comment (thc202) [inline org.zaproxy.ZAP.json:49]: Revert this as well, we have our own release workflow which will take care of this.

Comment (prodrigestivill) [reply org.zaproxy.ZAP.json]: Reverted, :)

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**